src/main/java/com/revature/services/EvalLogicImpl.java

```diff
@@ -16,66 +24,260 @@
 	@Autowired
 	private EvalRepository dao;
 	
+	@Autowired
+	private EvalCommentRepository commentDao;
+
+    @Autowired
+    private QuestionLogic questionLogic;
+
+    @Autowired
+    private JsonValidation validation;
+	
+	@PersistenceContext
+    private EntityManager entityManager;
+	
 
 //CREATE----------------------------
+	@Override
+	public Eval createEval(Eval eval) {
+		
+		System.out.println(eval);
+		
+		validation.validateEvalFields(eval);
+
+        Date date = new Date((new java.util.Date()).getTime());
+		eval.setDate(date);
+		
+		// Set eval property of questions and qEval property of question comments
+		// This is required because these fields are not auto-generated in json mapping
+		if (eval.getQuestions() != null) {
+			eval.getQuestions().forEach((question) -> {
+				question.setEval(eval);
+				validation.validateQuestionEvaluationFields(question);
+
+                // also set qEval of question comments
+				if (question.getComments() != null) {
+					question.getComments().forEach((comment) -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "comment" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1611)